x-pack/plugins/ml/server/lib/ml_client/ml_client.ts

```diff
@@ -491,7 +491,6 @@ export function getMlClient(
       return mlClient.startTrainedModelDeployment(...p);
     },
     async updateTrainedModelDeployment(...p: Parameters<MlClient['updateTrainedModelDeployment']>) {
-      await modelIdsCheck(p);
```

Comment:
These calls to `modelIdsCheck` are needed to restrict access to models that are in the current space only. 
Without them a user can update, stop and infer a model from spaces they do not have access to.

Reply:
The problem here with deployment IDs. We don't create saved objects for them, because they only live in deployment stats

Reply:
Thinking about it more, we'll have to leave the path variable as `modelId` so it is compatible with the Elasticsearch API that it forwards the request on to.
So the real model ID will probably need to be called something else, but still added to the body so we can check the space.